skytemple_files/common/ppmdu_config/pmdsky_debug/loader.py

```diff
@@ -0,0 +1,308 @@
+    "BELLY_LOST_PER_TURN": ["BELLY_LOST_TURN"],
+    "BELLY_DRAIN_IN_WALLS_INT": ["BELLY_LOST_WTW"],
+    "BELLY_DRAIN_IN_WALLS_THOUSANDTHS": ["BELLY_LOST_WTW1000"],
+    "FIXED_ROOM_ENTITY_SPAWN_TABLE": ["ENTITY_SPAWN_TABLE"],
+    "FIXED_ROOM_REVISIT_OVERRIDES": ["FIXED_FLOOR_OVERRIDES"],
+    "FIXED_ROOM_ITEM_SPAWN_TABLE": ["ITEM_SPAWN_TABLE"],
+    "FIXED_ROOM_MONSTER_SPAWN_TABLE": ["MONSTER_SPAWN_TABLE"],
+    "NECTAR_IQ_BOOST": ["NECTAR_IQ_GAIN"],
+    "FIXED_ROOM_TILE_SPAWN_TABLE": ["TILE_SPAWN_TABLE"],
+}
+
+
+class DeprecatedPmd2BinarySymbol(Pmd2BinarySymbol):
+    _DEPRECATION_WARNING = "The symbol {} is deprecated, use {} instead."
+
+    def __init__(self, name: str, begin: int, end: Optional[int], description: str,
+                 typ: Optional[Pmd2BinarySymbolType], new_name: str):
+        self._begin: int = None  # type: ignore
+        self.new_name: str = new_name
+        super().__init__(name, begin, end, description, typ)
+
+    @property  # type: ignore
+    def begin(self):
+        warnings.warn(self.__class__._DEPRECATION_WARNING.format(self.name, self.new_name), DeprecationWarning)
+        return self._begin
+
+    @begin.setter
+    def begin(self, value):
+        self._begin = value
+
+    @property
+    def end(self):
+        warnings.warn(self.__class__._DEPRECATION_WARNING.format(self.name, self.new_name), DeprecationWarning)
+        return self._end
+
+
+@dataclass
+class _IncompleteBinary:
+    loadaddress: Optional[int] = None
+    length: Optional[int] = None
+    symbols: List[Pmd2BinarySymbol] = field(default_factory=list)
+    description: str = field(default_factory=str)
+
+    @classmethod
+    def get(cls, pool: Dict[str, '_IncompleteBinary'], name: str) -> '_IncompleteBinary':
+        if name in pool:
+            return pool[name]
+        newobj = _IncompleteBinary()
+        pool[name] = newobj
+        return newobj
+
+
+def _read_symbol(loadaddr: int, symbol_def: dict, typ: Pmd2BinarySymbolType, region: str) -> Optional[Pmd2BinarySymbol]:
+    assert isinstance(loadaddr, int)
+    if 'name' in symbol_def:
+        name = symbol_def['name']
+    else:
+        raise ValueError("Symbol is missing it's name.")
+    
+    if 'address' in symbol_def:
+        if region in symbol_def['address']:
+            if isinstance(symbol_def['address'][region], list):
+                begin = symbol_def['address'][region][0] - loadaddr
+            else:
+                begin = symbol_def['address'][region] - loadaddr
+        else:
+            return None
+    else:
+        raise ValueError(f"Symbol {name} is missing an address.")
+    
+    if 'description' in symbol_def:
+        description = symbol_def['description']
+    else:
+        description = ""
+    
+    if typ == Pmd2BinarySymbolType.FUNCTION:
+        return Pmd2BinarySymbol(
+            name=name,
+            typ=Pmd2BinarySymbolType.FUNCTION,
+            begin=begin,
+            end=None,
```

Comment:
Technically there's no reason why a function symbol can't have a length. I just didn't add any because it's a bit easier not to and lengths aren't generally useful for functions. But is there really a need for the `if-else`?

Reply:
I can get rid of the if-else, it was mostly useful during writing the function since it would have error'd if there was a data block without a length. But I can just ignore it now if it doesn't exist.